samples/NativeLibrary/Class1.cs

```diff
@@ -37,20 +37,36 @@ public static int WriteLine(IntPtr pString)
         }
 
         [UnmanagedCallersOnly(EntryPoint = "sumstring")]
-        public static IntPtr sumstring(IntPtr first, IntPtr second)
+        public static unsafe char* sumstring(char* first, char* second)
         {
             // Parse strings from the passed pointers 
-            string my1String = Marshal.PtrToStringAnsi(first);
```

Comment:
If you were to `Console.WriteLine(*first)` to print the first character of the passed string, you would see that there's actually a mismatch between the C `char` type and C# `char` type - one is typically 1 byte, the other is 2 bytes.

It doesn't matter here because this is casting to `IntPtr` before doing anything meaningful, but because of that `IntPtr` is is a better choice in the signature of the method too.